### ISSUE TITLE
iGame hangs when not opening some WHDLoad error windows

### ISSUE THREAD

**Issue by gustavosarmento** [MrZammler/iGame#237]
For months I have an issue with iGame, where some games crash back to the iGame window, and it's freezed, with no action possible.

Now I found what happens: when running the game (for example: Death or Glory) directly from workbench, it crashes at the same point, but then I can see the respective WHDLoad error window (on this specific case, "Exception Address Error ($200c)", but it happens with others too). 

Somehow, when running from iGame, some WHDLoad error windows aren't appearing, needing a system reboot.

My system is an A1200 with TF1260, KS 3.2.2, MUI 3.8 and latest iGame and WHDLoad as of today.

Hope there's a way to fix it.

**Comment by walkero-gr** [comment]
@gustavosarmento 
Thank you for your message. Can you please let me know:
- what version of the OS are you using?
- how much memory your system has and what kind of CPU? Is it an AGA or ECS/OCS system?
- Do you know the reason why this error is happening? Is it a missing kickstart or something else?
- How could I reproduce this failure?
- Is there an older version of iGame that didn't freeze after such an error?

**Comment by gustavosarmento** [comment]
Let's go:

- OS 3.2.2 (Workbench and Kickstart - no soft kick);
- 2 MB Chip, 128 MB Fast, 68060 (full), AGA;
- Enabling the NOCACHE tool type gets rid of the error. All needed Kickstarts on DEVS:Kickstarts;
- Try this game (http://ftp.grandis.nu/turran/FTP/Retroplay%20WHDLoad%20Packs/Commodore_Amiga_-_WHDLoad_-_Games/D/DeathOrGlory_v1.0_De.lha). It happens on my 060 system, so I don't know if it also happens on 040 or 030;
- This issue has happened for quite a while at least, but I can't pinpoint since when, unfortunately.

**Comment by walkero-gr** [comment]
@gustavosarmento 
Nice machine you have there. 
I will try to reproduce it and see how this can be avoided. Will let you know as soon as I have any news.

**Comment by walkero-gr** [comment]
@gustavosarmento Hello there.
These days I revisited this issue and the only thing I can imagine is that when WHDload is run from shell, which is the way iGame does execute it, it doesn't show that error message. Maybe you could try it by opening a shell inside the game folder and writing 
`whdload SLAVE=DeathOrGloryDE.slave PRELOAD`

Unfortunately, I couldn't reproduce the issue on my emulated systems, and I do not have a computer with an 060. I would appreciate it if you could make this test and see if the error window pops up.

Also, can you please let me know the whdload version you are using?

**Comment by gustavosarmento** [comment]
Well, it just got weirder. I have now 3 behaviors:

- through iGame: it now executes the Startup script and stays on a black screen. Doesn't even give an error;
- through Shell: behaves as described initially;
- through Workbench: now the game runs!

My system: A1200 with TF1260, KS 3.2.3, MUI 3.8, iGame 2.4.6 and WHDLoad 19.2.

**Comment by walkero-gr** [comment]
> through iGame: it now executes the Startup script and stays on a black screen. Doesn't even give an error;

I guess you mean a black screen while it starts a game. If that's the case, this has nothing to do with iGame, because iGame's job is to run Whdload through shell. If that is run, iGame's job is finished.

I would suggest checking if you are connected online. The behaviour you described reminds me that you are connected to some network and Whdload doesn't like that.

**Comment by gustavosarmento** [comment]
You're right: I found out that the network shutdown scripts aren't 100% effective, at least not when running from Shell and/or iGame. If I eliminate completely the network connection (commenting out on User-startup), the game runs on these two instances as well.

Now, why it works through Workbench without any issues, just relying on the network shutdown scripts...

**Comment by walkero-gr** [comment]
I wonder that as well. Maybe there is a bug in WHDLoad when run from shell to not use these scripts. I will investigate it further.

Thank you again for those tests

**Comment by walkero-gr** [comment]
I just did a quick test, and it seems that whdload executes correctly the "StartUp" and "CleanUp" scripts, even when it is run from the shell. So, it might be something with the scripts you are using.

**Comment by gustavosarmento** [comment]
I thought about it. Problem is: they're exactly the same Startup and Cleanup scripts, be it run from WB, Shell or iGame (after all, they're specified on WHDLoad.prefs in only one instance each: ExecuteStartup and ExecuteCleanup). And yet, when running from Workbench, they work perfectly.

**Comment by walkero-gr** [comment]
Yeah, I agree with you. It is bizarre that the result is so different.

**Comment by gustavosarmento** [comment]
One eternity later... I found the issue, while testing WHDLoad 20 beta:

Somehow, even after issuing the Startup script, my PCMCIA network card kept sending level 2 interrupt signals for the Amiga. Apparently, WHDLoad 19 didn't care for it when ran directly, but through iGame it did.

Now WHDLoad 20 hanged on the same situation, with or without iGame.

The solution was to add another command (CardReset FORCE) to my startup script, and now everything is as it should be.

Sorry for taking your time with this, but at least you can now close this issue!

Btw, any plans for a new iGame version?

**Comment by walkero-gr** [comment]
> Sorry for taking your time with this, but at least you can now close this issue!

No need to be sorry. Actually, I am sorry I couldn't help more with it.

> Btw, any plans for a new iGame version?

Absolutely. I already work on a new version. I am hunting a bug that some people mentioned of double items in the list. It is difficult because I still can't find a way to reproduce it. But I am on it.

Thank you for getting back, and letting me know about the solution. Take care.
